### PR TITLE
docs: Add several docs and re-organize

### LINKED ISSUE #1088: Docs: document the detail of Sensor/EventBus message processing logic.
**Is your feature request related to a problem? Please describe.**
Document the detail of how EventBus processes events such as:

- single dependency
- multiple dependencies
- failures
- etc

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by whynowy**
Signed-off-by: Derek Wang <whynowy@gmail.com>

Closes #1088, #1074 and #1075

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by whynowy** [comment]
Missing POD priority for EventSource and Sensor..

File: docs/dr_ha_recommendations.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+[nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector)
+and
+[tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
+also could be set through `spec.nats.native.nodeSelector` and
+`spec.nats.native.tolerations`.
+
+### POD Priority
+
+Setting
+[POD Prioryty](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/)
```
Comment (alexec) [inline docs/dr_ha_recommendations.md:95]: spelling

**Comment by whynowy** [comment]
> Missing POD priority for EventSource and Sensor..

Added.

File: mkdocs.yml [hunk 2]
```diff
@@ -56,30 +39,53 @@ nav:
...
-      - 'triggers/aws-lambda.md'
-      - 'triggers/http-trigger.md'
-      - 'triggers/nats-trigger.md'
-      - 'triggers/kafka-trigger.md'
-      - 'triggers/k8s-object-trigger.md'
-      - 'triggers/log.md'
-      - 'triggers/openwhisk-trigger.md'
-      - 'triggers/slack-trigger.md'
-      - 'triggers/build-your-own-trigger.md'
+  - Sensors:
```
Comment (alexec) [inline mkdocs.yml:70]: please make sure your run `mkdocs serve` and it all looks good

**Review by alexec — APPROVED** [review]

File: docs/dr_ha_recommendations.md (hunk 1 above)
Comment (whynowy) [reply docs/dr_ha_recommendations.md]: Thanks, fixed!

**Review by whynowy — COMMENTED** [review]
...

**Review by alexec — APPROVED** [review]

**Pull request merged**